assessments/projects/contour/self-assessment.md

```diff
@@ -0,0 +1,206 @@
+# Contour Self-assessment
+
+## Table of contents
+
+* [Metadata](#metadata)
+  * [Security links](#security-links)
+* [Overview](#overview)
+  * [Actors](#actors)
+  * [Actions](#actions)
+  * [Background](#background)
+  * [Goals](#goals)
+  * [Non-goals](#non-goals)
+* [Self-assessment use](#self-assessment-use)
+* [Security functions and features](#security-functions-and-features)
+* [Project compliance](#project-compliance)
+* [Secure development practices](#secure-development-practices)
+* [Security issue resolution](#security-issue-resolution)
+* [Appendix](#appendix)
+
+## Metadata
+
+|   |  |
+| -- | -- |
+| Software | https://github.com/projectcontour/contour |
+| Security Provider | No |
+| Languages | Go |
+| SBOM | Contour does not currently produce an SBOM at build time |
+| | |
+
+### Security links
+
+| Doc | url |
+| -- | -- |
+| Security file | https://github.com/projectcontour/contour/blob/main/SECURITY.md |
+| Default and optional configs | [Contour configuration](https://github.com/projectcontour/contour/blob/main/internal/contourconfig/contourconfiguration.go) <br /> https://projectcontour.io/docs/1.27/  |
+
+## Overview
+
+Contour is a Kubernetes ingress control providing the control plane for the Envoy edge and service proxy. Contour supports dynamic configuration updates and multi-team ingress-delegation out of the box, while maintaining a lightweight profile.
+
+### Background
+
+As an ingress controller, Contour manages incoming traffic for services hosted via Kubernetes clusters. Built on Envoy, an intermediary edge proxy between clients and services, Contour provides functionalities to route traffic to the appropriate services based on defined rules. 
+
+While Kubernetes already has base resources for ingress control, Contour utilizes those resources and extends Kubernetes’ capabilities for managing external access to services running within a cluster.
+
+### Actors
+There are three separate actors in this project’s system:
+- **_Contour_**, which acts as a management server for Envoy and provides it with configuration. It interacts with the external actor, Envoy.
+
+### Actions
+For Contour to serve as an ingress controller it must be deployed along with Envoy, the external actor, as separate containers in Kubernetes. Contour is a client of the Kubernetes API and watches it utilizing controller-runtime primitives. In the Envoy Pods, Contour runs as an init container and writes the configuration to a temporary volume. After initialization is complete, the Envoy container starts and retrieves the bootstrap configuration written by Contour.
+When a request is sent, first it is routed by a load-balancer to an instance of Envoy. The Envoy proxy then sends the request to a Contour pod.
+
+
+### Goals
+Contour serves as a Layer 7 HTTP middleware reverse proxy for enabling ingress to Kubernetes clusters.
```

Comment:
Can you elaborate on the goals of this project? Typically, we have 3-4 goals at the least.

Reply:
We've elaborated on and added a few more goals for the project